src/hotspot/share/utilities/growableArray.hpp

```diff
@@ -209,13 +209,13 @@ class GrowableArrayView : public GrowableArrayBase {
     return -1;
   }
 
-  int  find(void* token, bool f(void*, E)) const {
+  template<typename T>
+  int  find(T* token, bool f(T*, E)) const {
```

Comment:
Pardon my ignorance here, but what is the type relationship between T and E?

Reply:
It's arbitrary and chosen by the caller through `f`, so I can't say :-). The best use case we have now is when you only have an `int` which uniquely describes an `LGRPSpace`.

Reply:
This code looks similar to a capturing lambda. Would an alternative be to use that instead and let the calling code be changed to:

```
  int i = lgrp_spaces()->find([](LGRPSpace* space) {
    return space->lgrp_id() == lgrp_id;
  });
```

Alternatively:
```
  auto matches_lgrp_id = [](LGRPSpace* space) {
    return space->lgrp_id() == lgrp_id;
  };
  
  int i = lgrp_spaces()->find(matches_lgrp_id);
```

Reply:
We could just as well do a capturing lambda here, yes. Then we'd have:

```c++
template<F>
int find(F finder);
```

It'd be a template instead of function pointer since it's a capturing lambda and  `std::function` is not permitted in Hotspot AFAIK. 

As an aside, to clarify for readers: There's a `&` missing in the capture list of your examples.

Reply:
Updated in-place

Reply:
Not a review, just agreeing with @stefank and @jdksjolen .  What they describe is idiomatic C++.

Reply:
Also, why isn't this change also being applied to `find_from_end`

Reply:
There can be a confusion related to selection of type names T and E:
  T is intuitively treated as a table and E as an element.
No pressure but I wonder if using D instead of T would be better.

Reply:
> Also, why isn't this change also being applied to `find_from_end`

Thank you @kimbarrett, the function is also changed accordingly.

Reply:
> We could just as well do a capturing lambda here, yes. Then we'd have:
> 
> ```c++
> template<F>
> int find(F finder);
> ```
> 
> It'd be a template instead of function pointer since it's a capturing lambda and `std::function` is not permitted in Hotspot AFAIK.
> 
> As an aside, to clarify for readers: There's a `&` missing in the capture list of your examples.

It would be nice to have this templated Function as finder. However, I think it is better to keep the changes small and manageable for this PR. Thanks for the comment.

Reply:
> Also, why isn't this change also being applied to `find_from_end`

Thanks Kim! It is also changed.